src/lib/utils/settings/abstract_setting.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+namespace opossum {
+
+/**
+ * This is an abstract class for all settings objects.
+ *
+ * Settings register itself at the SettingsManager.
+ * Therefore it is neccessary that they are owned by a shared pointer (instantiate with std::make_shared).
+ * Thus, you need to call unenroll() when the setting is not longer needed (it won't get destructed otherwise,
+ * as SettingsManager would still hold a shared pointer to it).
```

Comment:
This sounds a lot like a `unique_ptr` to me.

Reply:
Can I also get something like `shared_from_this()` on the object to auto-register settings at the settings manager after `std::make_unique<>()`?

Reply:
Ah, I was on the wrong track. Does the comment need the technical justification? Maybe just "settings need to be registered and deregistered in the SettingsManager. If a plugin is unloaded and does not remove its settings, this might lead to undefined behavior.'?